rtl/eclrtl/eclrtl.hpp

```diff
@@ -843,14 +843,32 @@ interface IEmbedFunctionContext : extends IInterface
     virtual void paramWriterCommit(IInterface *writer)=0;
     virtual void writeResult(IInterface *esdl, const char *esdlservice, const char *esdltype, IInterface *writer)=0;
     virtual void loadCompiledScript(size32_t len, const void *script) = 0;
+    // If reusing a context, need to call these before using/after using
+    virtual void enter() = 0;
```

Comment:
I'm not sure if enter() is the best name - I assumed from reading the code that it was a critical section.

Reply:
It's not THAT different a concept from a critical section... It's a local stack frame